### ISSUE TITLE
Crashing on laod

### ISSUE THREAD

**Issue by Godwhitelight** [mega12345mega/NBT-Editor#46]
```info: CONNECTING TO SERVER undefined
info: [2023-05-09T00:22:20Z INFO  rust_launcher::start::bindings] server: None
info: [2023-05-09T00:22:20Z INFO  rust_launcher::start] Loading version
info: {"versionSize":221129687,"state":{"VersionLoading":true},"currentDownloadRate":0,"expectedTasks":6,"downloadSpeed":0,"downloaded":0}
info: [2023-05-09T00:22:20Z INFO  rust_launcher::start] Finished loading version
info: [2023-05-09T00:22:20Z INFO  rust_launcher::start] Finished moving mods
info: [2023-05-09T00:22:20Z INFO  rust_launcher::start] Preparing assets
[2023-05-09T00:22:20Z INFO  rust_launcher::start] Preparing JRE
info: {"versionSize":221129687,"state":{"Downloading":["Assets","Jre","Libraries","Mods"]},"currentDownloadRate":0,"expectedTasks":6,"downloadSpeed":0,"downloaded":79183}
info: [2023-05-09T00:22:20Z INFO  rust_launcher::start] Preparing libraries
info: {"versionSize":221129687,"state":{"Downloading":["Assets","Libraries","Mods"]},"currentDownloadRate":0,"expectedTasks":6,"downloadSpeed":0,"downloaded":43227064}
info: [2023-05-09T00:22:20Z INFO  rust_launcher::start] Preparing mods
info: [2023-05-09T00:22:20Z INFO  rust_launcher::start] Finished preparing mods
info: {"versionSize":221129687,"state":{"Downloading":["Assets","Libraries"]},"currentDownloadRate":0,"expectedTasks":6,"downloadSpeed":0,"downloaded":47389133}
info: [2023-05-09T00:22:21Z INFO  rust_launcher::start] Finished preparing assets
info: {"versionSize":221129687,"state":{"Downloading":["Assets","Libraries"]},"currentDownloadRate":0,"expectedTasks":6,"downloadSpeed":0,"downloaded":708970319}
info: {"versionSize":221129687,"state":{"Downloading":["Libraries"]},"currentDownloadRate":0,"expectedTasks":6,"downloadSpeed":0,"downloaded":708970319}
info: [2023-05-09T00:22:21Z INFO  rust_launcher::start] Finished preparing libraries
[2023-05-09T00:22:21Z INFO  rust_launcher::start] Preparing registry
info: [2023-05-09T00:22:21Z INFO  rust_launcher::start] Finished preparing registry
info: {"versionSize":221129687,"state":{"Jvm":true},"currentDownloadRate":1707,"eta":"4 days","expectedTasks":6,"downloadSpeed":1707,"downloaded":802986689}
info: [2023-05-09T00:22:21Z INFO  rust_launcher::common::launch] starting JVM with arguments: ["-Djava.library.path=libraries/native\\net/digitalingot/fcef-windows/0.0.5\\extracted/;libraries/native\\net/digitalingot/fwebp-windows/0.0.1\\extracted/;libraries/native\\de/jcm/discord-game-sdk4j-windows/0.5.4\\extracted/;libraries/native\\org/jitsi/jnopus/1.0\\extracted/;libraries/native\\org/lwjgl/lwjgl/3.3.1\\extracted/;libraries/native\\org/lwjgl/lwjgl-jemalloc/3.3.1\\extracted/;libraries/native\\org/lwjgl/lwjgl-openal/3.3.1\\extracted/;libraries/native\\org/lwjgl/lwjgl-opengl/3.3.1\\extracted/;libraries/native\\org/lwjgl/lwjgl-glfw/3.3.1\\extracted/;libraries/native\\org/lwjgl/lwjgl-stb/3.3.1\\extracted/;libraries/native\\org/lwjgl/lwjgl-tinyfd/3.3.1\\extracted/;libraries/native\\com/mojang/text2speech/1.13.9\\extracted/", "--add-opens=java.desktop/java.awt.event=ALL-UNNAMED", "--add-opens=java.desktop/java.awt.color=ALL-UNNAMED", "--add-opens=java.desktop/java.awt=ALL-UNNAMED", "--add-opens=java.base/java.lang=ALL-UNNAMED", "-Xmx2048M", "-XX:+UnlockExperimentalVMOptions", "-XX:+UseG1GC", "-XX:G1NewSizePercent=20", "-XX:G1ReservePercent=20", "-XX:MaxGCPauseMillis=50", "-XX:G1HeapRegionSize=32M", "-Dlog4j2.formatMsgNoLookups=true", "-XX:ErrorFile=feather/java_error.log", "-Djavax.accessibility.assistive_technologies=", "-Djavax.net.ssl.trustStoreType=WINDOWS-ROOT", "-cp", "libraries\\java\\net/minecraft/client/1.19.4/minecraft-1.19.4.jar;libraries\\java\\net/fabricmc/tiny-mappings-parser/0.3.0+build.17/tiny-mappings-parser-0.3.0+build.17.jar;libraries\\java\\net/fabricmc/sponge-mixin/0.12.4+mixin.0.8.5/sponge-mixin-0.12.4+mixin.0.8.5.jar;libraries\\java\\net/fabricmc/tiny-remapper/0.8.2/tiny-remapper-0.8.2.jar;libraries\\java\\net/fabricmc/access-widener/2.1.0/access-widener-2.1.0.jar;libraries\\java\\org/ow2/asm/asm/9.4/asm-9.4.jar;libraries\\java\\org/ow2/asm/asm-analysis/9.4/asm-analysis-9.4.jar;libraries\\java\\org/ow2/asm/asm-commons/9.4/asm-commons-9.4.jar;libraries\\java\\org/ow2/asm/asm-tree/9.4/asm-tree-9.4.jar;libraries\\java\\org/ow2/asm/asm-util/9.4/asm-util-9.4.jar;libraries\\java\\net/fabricmc/intermediary/1.19.4/intermediary-1.19.4.jar;libraries/java\\net/fabricmc/fabric-loader/0.14.19/fabric-loader-0.14.19.jar;libraries\\java\\de/javagl/obj/0.3.0/obj-0.3.0.jar;libraries\\java\\de/jcm/discord-game-sdk4j/0.5.4/discord-game-sdk4j-0.5.4;libraries\\java\\net/digitalingot/feather-server-api-proto/1.1-SNAPSHOT/feather-server-api-proto-1.1-SNAPSHOT.jar;libraries\\java\\net/digitalingot/rust-extension/1.0.7/rust-extension-1.0.7.jar;libraries\\java\\net/digitalingot/fcef/0.0.5/fcef-0.0.5.jar;libraries\\java\\net/digitalingot/fwebp/0.0.1/fwebp-0.0.1.jar;libraries\\java\\net/digitalingot/feather-server-api/messaging/0.0.2/messaging-0.0.2-SNAPSHOT.jar;libraries\\java\\net/digitalingot/feather-server-api/api/0.0.2/api-0.0.2-SNAPSHOT.jar;libraries\\java\\org/jitsi/libjitsi/1.0-CUSTOM/libjitsi-1.0-CUSTOM.jar;libraries\\java\\org/jitsi/jitsi-utils/1.0-45-gc3afb76/jitsi-utils-1.0-45-gc3afb76.jar;libraries\\java\\org/capnproto/runtime/0.1.10/runtime-0.1.10.jar;libraries\\java\\com/google/inject/guice/5.1.0/guice-vendored-5.1.0.jar;libraries\\java\\javassist/javassist/3.12.1.GA/javassist-3.12.1.GA.jar;libraries\\java\\com/github/oshi/oshi-core/6.2.2/oshi-core-6.2.2.jar;libraries\\java\\com/google/code/gson/gson/2.10/gson-2.10.jar;libraries\\java\\com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar;libraries\\java\\com/google/guava/guava/31.1-jre/guava-31.1-jre.jar;libraries\\java\\com/ibm/icu/icu4j/71.1/icu4j-71.1.jar;libraries\\java\\com/mojang/authlib/3.18.38/authlib-3.18.38.jar;libraries\\java\\com/mojang/blocklist/1.0.10/blocklist-1.0.10.jar;libraries\\java\\com/mojang/brigadier/1.0.18/brigadier-1.0.18.jar;libraries\\java\\com/mojang/datafixerupper/6.0.6/datafixerupper-6.0.6.jar;libraries\\java\\com/mojang/javabridge/2.0.25/javabridge-2.0.25.jar;libraries\\java\\com/mojang/logging/1.1.1/logging-1.1.1.jar;libraries\\java\\com/mojang/patchy/2.2.10/patchy-2.2.10.jar;libraries\\java\\com/mojang/text2speech/1.13.9/text2speech-1.13.9.jar;libraries\\java\\commons-codec/commons-codec/1.15/commons-codec-1.15.jar;libraries\\java\\commons-io/commons-io/2.11.0/commons-io-2.11.0.jar;libraries\\java\\commons-logging/commons-logging/1.2/commons-logging-1.2.jar;libraries\\java\\io/netty/netty-codec-http/4.1.82.Final/netty-codec-http-4.1.82.Final.jar;libraries\\java\\io/netty/netty-buffer/4.1.82.Final/netty-buffer-4.1.82.Final.jar;libraries\\java\\io/netty/netty-codec/4.1.82.Final/netty-codec-4.1.82.Final.jar;libraries\\java\\io/netty/netty-common/4.1.82.Final/netty-common-4.1.82.Final.jar;libraries\\java\\io/netty/netty-handler/4.1.82.Final/netty-handler-4.1.82.Final.jar;libraries\\java\\io/netty/netty-resolver/4.1.82.Final/netty-resolver-4.1.82.Final.jar;libraries\\java\\io/netty/netty-transport-classes-epoll/4.1.82.Final/netty-transport-classes-epoll-4.1.82.Final.jar;libraries\\java\\io/netty/netty-transport-native-unix-common/4.1.82.Final/netty-transport-native-unix-common-4.1.82.Final.jar;libraries\\java\\io/netty/netty-transport/4.1.82.Final/netty-transport-4.1.82.Final.jar;libraries\\java\\it/unimi/dsi/fastutil/8.5.9/fastutil-8.5.9.jar;libraries\\java\\net/java/dev/jna/jna-platform/5.12.1/jna-platform-5.12.1.jar;libraries\\java\\net/java/dev/jna/jna/5.12.1/jna-5.12.1.jar;libraries\\java\\net/sf/jopt-simple/jopt-simple/5.0.4/jopt-simple-5.0.4.jar;libraries\\java\\org/apache/commons/commons-compress/1.21/commons-compress-1.21.jar;libraries\\java\\org/apache/commons/commons-lang3/3.12.0/commons-lang3-3.12.0.jar;libraries\\java\\org/apache/httpcomponents/httpclient/4.5.13/httpclient-4.5.13.jar;libraries\\java\\org/apache/httpcomponents/httpcore/4.4.15/httpcore-4.4.15.jar;libraries\\java\\org/apache/logging/log4j/log4j-api/2.19.0/log4j-api-2.19.0.jar;libraries\\java\\org/apache/logging/log4j/log4j-core/2.19.0/log4j-core-2.19.0.jar;libraries\\java\\org/apache/logging/log4j/log4j-slf4j2-impl/2.19.0/log4j-slf4j2-impl-2.19.0.jar;libraries\\java\\org/joml/joml/1.10.5/joml-1.10.5.jar;libraries\\java\\org/lwjgl/lwjgl-glfw/3.3.1/lwjgl-glfw-3.3.1.jar;libraries\\java\\org/lwjgl/lwjgl-jemalloc/3.3.1/lwjgl-jemalloc-3.3.1.jar;libraries\\java\\org/lwjgl/lwjgl-openal/3.3.1/lwjgl-openal-3.3.1.jar;libraries\\java\\org/lwjgl/lwjgl-opengl/3.3.1/lwjgl-opengl-3.3.1.jar;libraries\\java\\org/lwjgl/lwjgl-stb/3.3.1/lwjgl-stb-3.3.1.jar;libraries\\java\\org/lwjgl/lwjgl-tinyfd/3.3.1/lwjgl-tinyfd-3.3.1.jar;libraries\\java\\org/lwjgl/lwjgl/3.3.1/lwjgl-3.3.1.jar;libraries\\java\\org/slf4j/slf4j-api/2.0.1/slf4j-api-2.0.1.jar", "net.digitalingot.rustextension.ProxiedStart", "net.fabricmc.loader.launch.knot.KnotClient", "--username", "G0dwhitelight", "--version", "1.19.4-feather", "--gameDir", "C:\\Users\\bargi\\AppData\\Roaming/.minecraft", "--assetsDir", "assets", "--assetIndex", "1.19.4-feather-game", "--uuid", "32696a5c31b141a88df7e60369497e07", "--accessToken", "<hidden>", "--userType", "msa", "--versionType", "feather"]
info: sending javaw pid
info: starting cleanup
info: [03:22:22] [main/INFO]: Loading Minecraft 1.19.4 with Fabric Loader 0.14.19
info: [03:22:23] [main/INFO]: Loading 56 mods:
	- controlling 11.0.0.2
	- fabric-api 0.78.0+1.19.4
	   |-- fabric-api-base 0.4.23+9ff28bcef4
	   |-- fabric-api-lookup-api-v1 1.6.24+49abcf7ef4
	   |-- fabric-biome-api-v1 13.0.6+348a9c64f4
	   |-- fabric-block-api-v1 1.0.5+e022e5d1f4
	   |-- fabric-blockrenderlayer-v1 1.1.33+c2e6f674f4
	   |-- fabric-client-tags-api-v1 1.0.14+1134c5b8f4
	   |-- fabric-command-api-v1 1.2.26+f71b366ff4
	   |-- fabric-command-api-v2 2.2.5+e719b857f4
	   |-- fabric-commands-v0 0.2.43+df3654b3f4
	   |-- fabric-containers-v0 0.1.53+df3654b3f4
	   |-- fabric-content-registries-v0 3.5.8+ae0966baf4
	   |-- fabric-convention-tags-v1 1.4.0+9a7c5daaf4
	   |-- fabric-crash-report-info-v1 0.2.14+aeb40ebef4
	   |-- fabric-data-generation-api-v1 11.3.7+5da15ca1f4
	   |-- fabric-dimensions-v1 2.1.44+7f87f8faf4
	   |-- fabric-entity-events-v1 1.5.12+e45f7c65f4
	   |-- fabric-events-interaction-v0 0.4.42+a1ccd7bff4
	   |-- fabric-events-lifecycle-v0 0.2.51+df3654b3f4
	   |-- fabric-game-rule-api-v1 1.0.32+a1ccd7bff4
	   |-- fabric-item-api-v1 2.1.16+09a3510cf4
	   |-- fabric-item-group-api-v1 3.0.4+043f9acff4
	   |-- fabric-key-binding-api-v1 1.0.32+c477957ef4
	   |-- fabric-keybindings-v0 0.2.30+df3654b3f4
	   |-- fabric-lifecycle-events-v1 2.2.14+5da15ca1f4
	   |-- fabric-loot-api-v2 1.1.26+75e98211f4
	   |-- fabric-loot-tables-v1 1.1.30+9e7660c6f4
	   |-- fabric-message-api-v1 5.1.0+1ee8be40f4
	   |-- fabric-mining-level-api-v1 2.1.38+49abcf7ef4
	   |-- fabric-models-v0 0.3.29+11ba9c3bf4
	   |-- fabric-networking-api-v1 1.3.0+a6f3ccfaf4
	   |-- fabric-networking-v0 0.3.40+df3654b3f4
	   |-- fabric-object-builder-api-v1 7.0.2+63b515f4f4
	   |-- fabric-particles-v1 1.0.22+f1e4495bf4
	   |-- fabric-recipe-api-v1 1.0.7+a1ccd7bff4
	   |-- fabric-registry-sync-v0 2.1.3+a383ab97f4
	   |-- fabric-renderer-api-v1 2.2.4+81e8c576f4
	   |-- fabric-renderer-indigo 1.1.0+81e8c576f4
	   |-- fabric-renderer-registries-v1 3.2.37+df3654b3f4
	   |-- fabric-rendering-data-attachment-v1 0.3.27+afca2f3ef4
	   |-- fabric-rendering-fluids-v1 3.0.20+f1e4495bf4
	   |-- fabric-rendering-v0 1.1.40+df3654b3f4
	   |-- fabric-rendering-v1 2.1.0+8f878217f4
	   |-- fabric-resource-conditions-api-v1 2.3.0+e6c7d4eef4
	   |-- fabric-resource-loader-v0 0.11.1+1e1fb126f4
	   |-- fabric-screen-api-v1 1.0.44+8c25edb4f4
	   |-- fabric-screen-handler-api-v1 1.3.19+5da15ca1f4
	   |-- fabric-sound-api-v1 1.0.8+75e98211f4
	   |-- fabric-transfer-api-v1 3.1.0+d51205dbf4
	   \-- fabric-transitive-access-wideners-v1 3.0.2+63b515f4f4
	- fabricloader 0.14.19
	- feather 1.0.0-SNAPSHOT
	- java 17
	- minecraft 1.19.4
	- nbteditor 1.11.0
info: [03:22:23] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/C:/Users/bargi/AppData/Roaming/.minecraft/libraries/java/net/fabricmc/sponge-mixin/0.12.4+mixin.0.8.5/sponge-mixin-0.12.4+mixin.0.8.5.jar Service=Knot/Fabric Env=CLIENT
info: [03:22:23] [main/INFO]: Compatibility level set to JAVA_17
info: [03:22:24] [main/WARN]: Error loading class: com/terraformersmc/modmenu/config/option/EnumConfigOption (java.lang.ClassNotFoundException: com/terraformersmc/modmenu/config/option/EnumConfigOption)
info: [03:22:24] [main/WARN]: Error loading class: net/optifine/Config (java.lang.ClassNotFoundException: net/optifine/Config)
info: [03:22:26] [main/WARN]: Failed to add PDH Counter: \Paging File(_Total)\% Usage, Error code: 0xC0000BB8
info: [03:22:26] [main/WARN]: Failed to add counter for PDH counter: \Paging File(_Total)\% Usage
info: [03:22:26] [main/WARN]: Disabling further attempts to query Paging File.
info: [03:22:26] [main/WARN]: COM exception: Invalid Class Win32_PerfRawData_PerfOS_PagingFile
info: [03:22:34] [Render thread/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
info: [03:22:34] [Render thread/INFO]: Setting user: G0dwhitelight
info: [03:22:35] [Render thread/INFO]: [Indigo] Registering Indigo renderer!
info: [03:22:35] [Render thread/WARN]: COM exception: Invalid Class Win32_PerfRawData_PerfOS_PagingFile
info: ---- Minecraft Crash Report ----
// Don't be sad, have a hug! <3

Time: 2023-05-09 03:22:35
Description: Initializing game

java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors, provided by 'nbteditor'!
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at net.minecraft.class_310.<init>(class_310.java:468)
	at net.minecraft.client.main.Main.main(Main.java:198)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:462)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at net.digitalingot.rustextension.ProxiedStart.main(ProxiedStart.java:16)
Caused by: java.lang.ExceptionInInitializerError
	at com.luneruniverse.minecraft.mod.nbteditor.commands.factories.FactoryCommand.<init>(FactoryCommand.java:18)
	at com.luneruniverse.minecraft.mod.nbteditor.commands.factories.FactoryCommand.<clinit>(FactoryCommand.java:15)
	at com.luneruniverse.minecraft.mod.nbteditor.commands.CommandHandler.<clinit>(CommandHandler.java:43)
	at com.luneruniverse.minecraft.mod.nbteditor.NBTEditorClient.onInitializeClient(NBTEditorClient.java:50)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47)
	... 12 more
Caused by: java.lang.RuntimeException: Error invoking method
	at com.luneruniverse.minecraft.mod.nbteditor.multiversion.Reflection$MethodInvoker.invoke(Reflection.java:103)
	at com.luneruniverse.minecraft.mod.nbteditor.multiversion.Version.getReleaseTarget(Version.java:71)
	at com.luneruniverse.minecraft.mod.nbteditor.multiversion.Version.get(Version.java:20)
	at com.luneruniverse.minecraft.mod.nbteditor.multiversion.TextInst.translatable(TextInst.java:17)
	at com.luneruniverse.minecraft.mod.nbteditor.commands.factories.SignatureCommand.<clinit>(SignatureCommand.java:34)
	... 17 more
Caused by: java.lang.IllegalArgumentException: object is not an instance of declaring class
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.luneruniverse.minecraft.mod.nbteditor.multiversion.Reflection$MethodInvoker.invoke(Reflection.java:101)
	... 21 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at net.minecraft.class_310.<init>(class_310.java:468)

-- Initialization --
Details:
	Modules: 
		ADVAPI32.dll:Advanced Windows 32 Base API:10.0.22621.1554 (WinBuild.160101.0800):Microsoft Corporation
		COMCTL32.dll:������ ����� �� ������ �����:6.10 (WinBuild.160101.0800):Microsoft Corporation
		CRYPT32.dll:Crypto API32:10.0.22621.1554 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTBASE.dll:Base cryptographic API DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTSP.dll:Cryptographic Service Provider API:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		DBGHELP.DLL:Windows Image Helper:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		DNSAPI.dll:DNS Client API DLL:10.0.22621.1554 (WinBuild.160101.0800):Microsoft Corporation
		GDI32.dll:GDI Client DLL:10.0.22621.436 (WinBuild.160101.0800):Microsoft Corporation
		IMM32.DLL:Multi-User Windows IMM32 API Client DLL:10.0.22621.730 (WinBuild.160101.0800):Microsoft Corporation
		IPHLPAPI.DLL:IP Helper API:10.0.22621.1554 (WinBuild.160101.0800):Microsoft Corporation
		KERNEL32.DLL:Windows NT BASE API Client DLL:10.0.22621.885 (WinBuild.160101.0800):Microsoft Corporation
		KERNELBASE.dll:Windows NT BASE API Client DLL:10.0.22621.885 (WinBuild.160101.0800):Microsoft Corporation
		MSASN1.dll:ASN.1 Runtime APIs:10.0.22621.891 (WinBuild.160101.0800):Microsoft Corporation
		NSI.dll:NSI User-mode interface DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		NTASN1.dll:Microsoft ASN.1 API:10.0.22621.1554 (WinBuild.160101.0800):Microsoft Corporation
		OLEAUT32.dll:OLEAUT32.DLL:10.0.22621.436 (WinBuild.160101.0800):Microsoft Corporation
		Ole32.dll:Microsoft OLE for Windows:10.0.22621.1554 (WinBuild.160101.0800):Microsoft Corporation
		PSAPI.DLL:Process Status Helper:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		Pdh.dll:Windows Performance Data Helper DLL:10.0.22621.1554 (WinBuild.160101.0800):Microsoft Corporation
		RPCRT4.dll:Remote Procedure Call Runtime:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		SHCORE.dll:SHCORE:10.0.22621.1554 (WinBuild.160101.0800):Microsoft Corporation
		SHELL32.dll:Dll ����� �� ����� Windows:10.0.22621.1554 (WinBuild.160101.0800):Microsoft Corporation
		USER32.dll:DLL �� ���� USER API �� Windows ����� �������:10.0.22621.1554 (WinBuild.160101.0800):Microsoft Corporation
		USERENV.dll:Userenv:10.0.22621.1554 (WinBuild.160101.0800):Microsoft Corporation
		VCRUNTIME140.dll:Microsoft� C Runtime Library:14.27.29016.0 built by: vcwrkspc:Microsoft Corporation
		VERSION.dll:Version Checking and File Installation Libraries:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		WINHTTP.dll:Windows HTTP Services:10.0.22621.1554 (WinBuild.160101.0800):Microsoft Corporation
		WINMM.dll:MCI API DLL:10.0.22621.1554 (WinBuild.160101.0800):Microsoft Corporation
		WS2_32.dll:Windows Socket 2.0 32-Bit DLL:10.0.22621.1554 (WinBuild.160101.0800):Microsoft Corporation
		WSOCK32.dll:Windows Socket 32-Bit DLL:10.0.22621.1554 (WinBuild.160101.0800):Microsoft Corporation
		amsi.dll:Anti-Malware Scan Interface:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		antimalware_provider64.dll:AMSI provider```

**Comment by Godwhitelight** [comment]
this is feather fabric but i got the same result on normal fabric as well

**Comment by mega12345mega** [comment]
I'm not able to reproduce this crash. I'm not familiar with feather fabric; is the `rust_launcher` at the top a part of feather fabric, or a different thing? If you are using a custom launcher, it is possible it is doing something wonky with libraries. Based on the crash, this appears to be some sort of incompatibility with the `Version` system, which checks the current Minecraft version to determine the correct code to prevent crashes.
My best guess is that the "Mojang Bridge" library isn't right. If I remember correctly, it was removed and shouldn't be loaded for 1.19.4, but the `starting JVM with arguments` section of the log lists `com/mojang/javabridge`.
If you are using custom JVM arguments, you might be able to remove that item from the list. (Or try without any custom arguments.)
If you're wondering why this library is relevant, it used to contain `GameVersion` information, but was later merged in with the normal `GameVersion` class in Minecraft itself.

**Comment by Godwhitelight** [comment]
I have tried it with normal Minecraft fabric and I got the same result.
the rust_launcher is a part of feather fabric.
My guess is the JRE, so what JRE are you using?

**Comment by mega12345mega** [comment]
Java shouldn't be an issue, since it is bundled with the Minecraft launcher and a different incompatibility error would occur. The bridge library is definitely related, and this isn't affected by the Java version.
Can you confirm that you are launching with default JVM arguments? If you can get Minecraft to launch without the bridge library, that should fix it. (Unless that causes a different type of crash.)

Tonight I can compile a "fixed" version of the mod that you can test, but it would only work in 1.19.4.

Here's how the bridge library is related (simplified):

* Older Minecraft versions:
  * Bridge library exists
  * Minecraft's `GameVersion` extends Bridge's `GameVersion`
  * Bridge's `GameVersion` includes `getId`, which is accessible from Minecraft's, since it extends
* Newer Minecraft versions:
  * Bridge library doesn't exist (I'm unclear if it just doesn't contain a `GameVersion`, but I'm pretty sure it is completely gone)
  * Minecraft's `GameVersion` doesn't extend Bridge's (non-existent) `GameVersion`
  * Minecraft's `GameVersion` includes its own `getId`

How NBT Editor's `Version` determines the Minecraft version:
1. Check if Bridge's `GameVersion#getId` exists
2. If it exists, call Bridge's `getId` on Minecraft's `GameVersion`
3. If it doesn't exist, call Minecraft's `getId` on Minecraft's `GameVersion`

If the bridge library exists when it shouldn't:
1. Bridge's `GameVersion#getId` exists
2. Call Bridge's `getId` on Minecraft's `GameVersion`
3. Minecraft's `GameVersion` doesn't extend Bridge's `GameVersion`, so `getId` is incompatible
4. Java detects this conflict and crashes

**Comment by mega12345mega** [comment]
Here's the "fixed" version. This changes `Version` to simply output "1.19.4" instead of getting it normally. Additionally, it prints out some basic debug information to confirm that the bridge library is the issue.
Here is what I get printed out:
```
[STDOUT]: NBTEDITOR DEBUG:
[STDOUT]:  - getReleaseTargetExists: false
[STDOUT]:  - getIdExists: false
[STDOUT]:  - Bridge_GameVersion: false
[STDOUT]:  - SharedConstants extends: <none>
```
Can you enable the console in the launcher settings and send what you get outputted? I'm guessing you're going to see this:
```
[STDOUT]: NBTEDITOR DEBUG:
[STDOUT]:  - getReleaseTargetExists: false
[STDOUT]:  - getIdExists: true
[STDOUT]:  - Bridge_GameVersion: true
[STDOUT]:  - SharedConstants extends: <none>
```
If so, then this is the problem - notice `getIdExists` is `true` while `SharedConstants's (Minecraft's) GameVersion` extends nothing.

If this is indeed what you see, can you confirm that you are launching with default JVM arguments? If you can get Minecraft to launch without the bridge library, that should fix it.

Please note that I'm not going to keep this as a permanent solution, so I'm not going to keep releasing "fixed" versions until I understand the root cause and can properly fix it.

[nbteditor-1.11.999.zip](https://github.com/mega12345mega/NBT-Editor/files/11459366/nbteditor-1.11.999.zip)

**Comment by Godwhitelight** [comment]
ok, it launched.
```
[20:08:01] [Render thread/INFO]: [STDOUT]: NBTEDITOR DEBUG:
[20:08:01] [Render thread/INFO]: [STDOUT]:  - getReleaseTargetExists: false
[20:08:01] [Render thread/INFO]: [STDOUT]:  - getIdExists: true
[20:08:01] [Render thread/INFO]: [STDOUT]:  - Bridge_GameVersion: true
[20:08:01] [Render thread/INFO]: [STDOUT]:  - SharedConstants extends: <none>
```

**Comment by mega12345mega** [comment]
While looking into the version system more when debugging a different issue, I discovered a much easier way to get the version information; I can load the `id` and sometimes `release_target` directly from a `version.json` file stored in the game's `jar` file. This completely bypasses all the complicated stuff relating to this issue. In theory, this should solve this issue in a much nicer manner, so I'll mark this as completed.

**Comment by mega12345mega** [comment]
I've just released v1.12.0, which includes the change mentioned in my last comment. If you could confirm that that does fix it, that would be great!

**Comment by mega12345mega** [comment]
Assuming that the fix worked, I'm going to close the issue now, since it has been a month.
